detections/endpoint/malicious_powershell_executed_as_a_service.yml

```diff
@@ -10,7 +10,7 @@ description: This detection is to identify the abuse the Windows SC.exe to execu
   malicious commands or payloads via PowerShell.
 search: ' `wineventlog_system` EventCode=7045 | eval l_Service_File_Name=lower(Service_File_Name)
   | regex l_Service_File_Name="powershell[.\s]|powershell_ise[.\s]|pwsh[.\s]|psexec[.\s]"
-  | regex l_Service_File_Name="-nop[rofile]*|-w[indowstyle]*\s+hid[den]*|-noe[xit]*|-enc[odedcommand]*"
+  | regex l_Service_File_Name="-nop[rofile\s]+|-w[indowstyle]*\s+hid[den]*|-noe[xit\s]+|-enc[odedcommand\s]+"
```

Comment:
@miskoSplunk Random question, would it make any difference to use eval statements over regex (for performance)? 
```
| eval NoProf=if(match(lower(CommandLine),"-nop"),1,0)
```

```
| eval Hidden=if(match(lower(CommandLine)," hidden") OR match(lower(CommandLine),"-windowstyle") OR match(lower(CommandLine), "-nonewwindow"),1,0)
```
```
| eval Encode=if(match(lower(CommandLine),"[A-Za-z0-9+\/]{44,}([A-Za-z0-9+\/]{4}|[A-Za-z0-9+\/]{3}=|[A-Za-z0-9+\/]{2}==)") OR match(lower(CommandLine),"frombase64string"),5,0)
```

Reply:
@MHaggis Apologies for the late reply. 
I'm not sure what would work faster, but I also think we're fine with performance - event 7045 is not at all frequent, so this won't go into regexes section much :)